### PR TITLE
core[minor]: Add document compressor class to core

### PR DESCRIPTION
**Pull request by bracesproul**

### CODE REVIEW

File: langchain-core/langchain.config.js [hunk 1]
```diff
@@ -36,7 +36,8 @@ export const config = {
     prompt_values: "prompt_values",
     runnables: "runnables/index",
     "runnables/remote": "runnables/remote",
-    retrievers: "retrievers",
+    retrievers: "retrievers/index",
+    "retrievers/document_compressors/base": "retrievers/document_compressors/base",
```
Comment (jacoblee93) [inline langchain-core/langchain.config.js:40]: None of the other entrypoints end with `/base`

**Comment by jacoblee93** [comment]
Small comment, and are we sure we're ready to commit to this?

**Comment by bracesproul** [comment]
> Small comment, and are we sure we're ready to commit to this?

It's already in core in the py repo, only reason to wait would be if we wanted to make breaking changes?

**Comment by jacoblee93** [comment]
Thanks!

**Pull request merged**